File: src/packets/PacketC2SAllPower.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x0C
 * All Power
 * Sets on/off state of all zones
 * Data:
 *      (Unsigned Char) On/Off
 */
class PacketC2SAllPower extends PacketC2S {
    parseData() {
        this._power = (this._buffer.readUInt8() == 1);
    }

    getPowered() {
        return this._power;
    }

    getID() {
        return PacketC2SAllPower.ID;
    }
}

PacketC2SAllPower.ID = 0x0C;

module.exports = PacketC2SAllPower;

File: src/RNetPi.Core/Packets/PacketC2SAllPower.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x0C
/// All Power
/// Sets on/off state of all zones
/// Data:
///     (Unsigned Char) On/Off
/// </summary>
public class PacketC2SAllPower : PacketC2S
{
    public const byte ID = 0x0C;

    public bool Powered { get; private set; }

    public PacketC2SAllPower(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        Powered = Reader.ReadByte() == 1;
    }
}